css/selectors/focus-in-focus-event-001.html

```diff
@@ -14,17 +14,29 @@
 <script>
 var input = document.querySelector('input');
 input.addEventListener('focus', function(e) {
-  var focusPseudo = document.querySelector(':focus');
-  var focusVisiblePseudo = document.querySelector(':focus-visible');
-  var focusWithinPseudo = document.querySelector(':focus-within');
   test(() => {
-    assert_equals(e.target, focusPseudo, "':focus' matches event.target");
+    try {
+      var focusPseudo = document.querySelector(':focus');
+      assert_equals(e.target, focusPseudo, "':focus' matches event.target");
+    } catch (error) {
+      assert_unreached("':focus'  is an invalid selector. SyntaxError: " + error);
+    }
```

Comment:
Instead of a `try...catch` it may be simpler to use `test_valid_selector` from css/support/parsing-testcommon.js

```js
test(() => {
  test_valid_selector(':focus');
  assert_equals(e.target, focusPseudo, "':focus' matches event.target");
}, "Checks that ':focus' pseudo-class matches inside 'focus' event handler");
```
And ditto for the others.

Reply:
I think that does more checks that what I need. And they'll be duplicated with other tests in `selectors/parsing/`.